pennylane_lightning/src/tests/lightning_apply_unittest.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright 2021 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "gtest/gtest.h"
+#include "../Apply.hpp"
+
+using std::vector;
+
+namespace test_apply {
+
+class getIndicesAfterExclusionTestFixture :public ::testing::TestWithParam<std::tuple<unsigned int, unsigned int>> {
+};
+
+TEST_P(getIndicesAfterExclusionTestFixture, CheckgetIndicesAfterExclusionResults) {
+    unsigned int index = std::get<0>(GetParam());
+    unsigned int qubits = std::get<1>(GetParam());
+
+    vector<vector<unsigned int>> inputs = {
+        {0, 1},
+        {2},
+        {1, 2},
+        {1, 2, 4},
+        {3, 2, 4},
+    };
+
+    vector<vector<unsigned int>> outputs = {
+        {2},
+        {0, 1},
+        {0},
+        {0, 3},
+        {0, 1, 5},
+    };
+
+    vector<unsigned int> input = inputs[index];
+    vector<unsigned int> output = outputs[index];
+    ASSERT_EQ(output, Pennylane::getIndicesAfterExclusion(input, qubits));
+}
+
+INSTANTIATE_TEST_SUITE_P (
+        getIndicesAfterExclusionTests,
+        getIndicesAfterExclusionTestFixture,
+        ::testing::Values(
```

Comment:
@antalszava thanks for this [comment](https://github.com/PennyLaneAI/pennylane-lightning/pull/80/files#r587621564)! I tried but then realized that `Combine` does the product over inputs, we want more like a `Zip`.

However, I wouldn't say it's worth worrying about much, given that we're just effectively testing `set(a) - set(b)`.